### PR TITLE
Cleaning the code

### PR DESCRIPTION
**Pull request by informalict**
Unit tests:
I wrote 50 unit tests with different scenarios for `EnsurePods` and `ensureImages` and those tests pass on my branch and on the temporary branch `testing/after-cleaning` (I created from master that only for testing because I had to export a few functions and create short helpful functions, we can assume that branch is the same like master. Compare testing branch with master: https://github.com/arangodb/kube-arangodb/compare/testing/after-cleaning)

Integration tests:
Full integration tests passed with image `arangodb/enterprise-preview:3.5.3`

### CODE REVIEW

File: pkg/deployment/deployment_test.go [hunk 1]
```diff
@@ -0,0 +1,2327 @@
+package deployment
```
Comment (ajanikow) [inline pkg/deployment/deployment_test.go]: Add header

File: pkg/deployment/resources/pod_creator_arangod.go [hunk 2]
```diff
@@ -0,0 +1,261 @@
+package resources
```
Comment (ajanikow) [inline pkg/deployment/resources/pod_creator_arangod.go]: Add header

File: pkg/deployment/resources/pod_creator_sync.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
+package resources
```
Comment (ajanikow) [inline pkg/deployment/resources/pod_creator_sync.go]: Add header

File: pkg/util/k8sutil/lifecycle.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
+package k8sutil
```
Comment (ajanikow) [inline pkg/util/k8sutil/lifecycle.go]: Add header

File: pkg/deployment/deployment_test.go (hunk 1 above)
Comment (informalict) [reply pkg/deployment/deployment_test.go]: added

File: pkg/deployment/resources/pod_creator_sync.go (hunk 3 above)
Comment (informalict) [reply pkg/deployment/resources/pod_creator_sync.go]: added

File: pkg/util/k8sutil/lifecycle.go (hunk 4 above)
Comment (informalict) [reply pkg/util/k8sutil/lifecycle.go]: added

**Pull request merged**